### PR TITLE
Document some bugs

### PR DESCRIPTION
**Pull request by kiliwily**

### CODE REVIEW

**Comment by ShinyDragonHunter** [comment]
When updating bugs_and_glitches.md, it’s worth explaining what the bug is and why it happens. This is akin to how it’s done in pokecrystal’s documentation on bugs and glitches.

**Comment by kiliwily** [comment]
The bug is already documented in bugs_and_glitches.md akin it is done in pokecrystal. I just added an alternative way to fix it.

File: src/metatile_behavior.c [hunk 1]
```diff
@@ -968,6 +968,10 @@ bool8 MetatileBehavior_IsDiveable(u8 metatileBehavior)
 
 bool8 MetatileBehavior_IsUnableToEmerge(u8 metatileBehavior)
 {
+    //BUG: The player is unintentionally able to emerge on water doors.
+    //Also the narrower underwater door in the underwater tileset has the wrong metatile behavior. This causes the dive glitch.
+    //To fix that add ||metatileBehavior == MB_WATER_DOOR to the if statement below and 
+    //change the metatile behavior of the narrower water door with porymaps tilset editor.
```
Comment (GriffinRichards) [inline src/metatile_behavior.c:974]: typo, `tilset`. Also space between `||` and `metatileBehavior`

File: src/battle_util.c [hunk 2]
```diff
@@ -527,6 +527,8 @@ void HandleAction_ThrowPokeblock(void)
         gBattleStruct->safariPkblThrowCounter++;
     if (gBattleStruct->safariEscapeFactor > 1)
     {
+        //BUG: The safariEscapeFactor is unintetionally able to become 0 (but it can not become negative!). This causes the pokeblock throw glitch.
```
Comment (GriffinRichards) [inline src/battle_util.c:530]: Put a space after `//`

**Review by GriffinRichards — CHANGES_REQUESTED** [review]
These are probably fine for now, but I'd prefer if these simple fixes were in directives rather than described. For example `//To fix this add another if statement after the instruction that sets currentHP = 1 if currentHP <= 0.` is unnecessarily wordy

**Comment by ShinyDragonHunter** [comment]
In one of the changes, you wrote `safariEscapeFactor is unintetionally`

**Comment by kiliwily** [comment]
Typos are fixed. Spaces are added after //. Pomeg fix is shortend.
Should I write the fixes as commented out code?

**Comment by PokeCodec** [comment]
> Typos are fixed. Spaces are added after //. Pomeg fix is shortend.
> Should I write the fixes as commented out code?

Yes

**Comment by GriffinRichards** [comment]
That's not necessary, at that point they should just be directives. Comments are fine for now

**Review by GriffinRichards — APPROVED** [review]

**Pull request merged**